src/odemis/odemisd/mdupdater.py

```diff
@@ -284,36 +284,36 @@ def updateLightPower(power, light=light, comp=comp):
 
         return True
 
-    def observeSpectrograph(self, spectrograph, comp):
-        if comp.role != "monochromator":
+    def observeSpectrograph(self, spectrograph, comp_affected):
+        if comp_affected.role != "monochromator":
             return False
 
         if 'slit-monochromator' not in spectrograph.axes:
             logging.info("No 'slit-monochromator' axis was found, will not be able to compute monochromator bandwidth.")
 
-            def updateOutWLRange(pos, sp=spectrograph, comp=comp):
-                wl = sp.position.value["wavelength"]
+            def updateOutWLRange(pos, comp_affected=comp_affected):
+                wl = pos["wavelength"]
                 md = {model.MD_OUT_WL: (wl, wl)}
-                comp.updateMetadata(md)
+                comp_affected.updateMetadata(md)
 
         else:
-            def updateOutWLRange(pos, sp=spectrograph, comp=comp):
+            def updateOutWLRange(pos, sp=spectrograph, comp_affected=comp_affected):
                 width = pos['slit-monochromator']
-                bandwidth = sp.getOpeningToWavelength(width)
+                bandwidth = sp.getOpeningToWavelength(width)  # TODO can you also replace spectrograph here somehow?
```

Comment:
No, there is no way to replace `sp` here. You need to call it. Just leave this as-is.